tofhir-engine/src/main/scala/io/tofhir/engine/model/MappingJobSourceSettings.scala

```diff
@@ -58,8 +58,11 @@ case class SqlSourceSettings(name: String, sourceUri: String, databaseUrl: Strin
  * @param sourceUri        Computer friendly canonical url indicating the source of the data (May be used for Resource.meta.source)
  * @param bootstrapServers Kafka bootstrap server(s) with port, may be comma seperated list (localhost:9092,localhost:9091)
  * @param asRedCap         Indicate whether it is a RedCap source
+ * @param batchRead        If true, enables batch reading mode (`read`), otherwise enables streaming mode (`readStream`).
+ *                         Batch mode (`read`) processes data as a one-time load, ideal for static datasets i.e. mapping testing functionality.
+ *                         Streaming mode (`readStream`) continuously ingests new data as it arrives, suitable for real-time data sources.
  */
-case class KafkaSourceSettings(name: String = "", sourceUri: String = "", bootstrapServers: String = "", asRedCap: Boolean = false) extends MappingJobSourceSettings {
+case class KafkaSourceSettings(name: String = "", sourceUri: String = "", bootstrapServers: String = "", asRedCap: Boolean = false, batchRead: Boolean = false) extends MappingJobSourceSettings {
```

Comment:
Let's use asStream instead of creating new property to avoid confusion

Reply:
Done.